### ISSUE TITLE
Do not run certain PR checks on forks and automated PRs

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3281]
Certain PR checks, such as deploying ZIPs or Storybook, don't work on forks of Site Kit, and they might also be unnecessary for e.g. automated PRs like [this from Dependabot](https://github.com/google/site-kit-wp/pull/3186). So such GitHub actions should be configured to not run in those scenarios.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eugene-manuilov** [comment]
> All Site Kit GitHub actions that deploy data to the Wiki or GitHub pages should be adjusted so that they no longer run for forks of the repository (since they would typically fail).

@aaemnnosttv @felixarntz both Zip and Storybook actions already don't run for forks. Do we need to rephrase it to say that it doesn't run for dependabot branches?

**Comment by tofumatt** [comment]
I think the Dependabot branches aren't _exactly_ forks because they're created by a `Bot` user and not a regular user who makes forks.

So yeah, ignoring `dependabot` PRs by using the branch name makes sense; they're all made on the actual Site Kit repo because it's a privileged GitHub bot. (See: https://github.com/google/site-kit-wp/tree/dependabot/npm_and_yarn/path-parse-1.0.7 from https://github.com/google/site-kit-wp/pull/3824)

IB ✅

**Comment by asvinb** [comment]
### QA: ✅ 

* Created dummy PR from a branch that starts with `dependabot/`
* The Zip and Storybook workflows do not run.
![image](https://user-images.githubusercontent.com/31818/135074668-1a96cbed-de23-4a1a-8022-3978f56a53d8.png)
